### DISCUSSION TITLE
Double-check of HVDC modeling assumptions

### DISCUSSION THREAD

**Discussion by ekatef** [pypsa-meets-earth/pypsa-earth#408] [General]
We have implemented two options for HVDC representation: as PyPSA Links components or as PyPSA Lines. The link representation follows the approach used in PyPSA-Europe. The line representation implies that we use a special case of the AC transmission line by setting the line parameters via the standard line type. The question I'm struggling a bit is how to set these parameters properly.

The Line component in PyPSA implements a classic [PI model](https://pypsa.readthedocs.io/en/latest/power_flow.html#line-model) with the line parameters defined via the active resistance `r`, the inductivity resistance `x` and the capacitance `c` (set per length unit). We adopt this model to represent the DC transmission lines by setting some artificial values for `x` and `c`: the `x` should be close to zero (exact `0` leads to numerical troubles) while `c` should be very big (if I understand it correctly).

The questions I'm not sure about:

1) How much are small enough and big enough for `x` and `c`, respectivelly? 

2) If I'm not mistaken there are some capacitance effects for DC lines as well. Does it make sense trying to account this? Or can we be sure that it can be neglected for the problems valid for PyPSA?

@davide-f, @pz-max, @Ekaterina-Vo I would be very grateful for looking into it :)

**Comment by pz-max** [comment]
Hi @ekatef , thanks for sharing this in the discussion. 
For DC lines, meaning `frequency=0`, the reactance `x` and capacitance `c` need to be zero (close to zero). From this [article](https://en.wikipedia.org/wiki/High-voltage_direct_current), DC powered cables are limited only by their temperature rise and [Ohm's law](https://en.wikipedia.org/wiki/Ohm%27s_Law) where only the resistance counts. Charging of the lines happens only once which is negligble.

To explain a bit more, please check the picture below (but forget about all the given names). When you have AC power, the reactance `x` and the capacitance `c` can lead to the case that the sinusoidal waveforms are shifted. Adding a little bit more reactance shifts the curves in one direction, while adding capacitance shifts the curves in the opposite direction. 

So what happens if you only consider DC power? You basically have only straight lines. No sinusoidal curves. So there is practically no influence of capacitance `c` or reactance `x` on the power flow -- nothing can be shifted. So it would be best setting both values `0` for DC lines in PyPSA. 

I think what would make sense is to import a "standard linetype" from PyPSA and set the value `x` and `c` to zero. But I remember you faced issues with this. Can you describe how we can reproduce the bug? Or post it on PyPSA as an issue? 

![image](https://user-images.githubusercontent.com/61968949/180434529-6e95e0a4-1012-4a05-aa5a-204cdfd03330.png)

**Reply by ekatef** [reply]
@pz-max, thank you so much for the comments! They were really helpful.

Absolutely agree that both the inductive reactance `x` and the capacity reactance `x_c` should be `0` for DC. But there is a little pitfall in definitions: `x_c` is inversely proportional to capacitance `c` and `x_c == 0` means that `c` is very big.

> From this [article](https://en.wikipedia.org/wiki/High-voltage_direct_current), DC powered cables are limited only by their temperature rise and [Ohm's law](https://en.wikipedia.org/wiki/Ohm%27s_Law) where only the resistance counts. Charging of the lines happens only once which is negligble.

Great, thank you! That's a relief to have a confirmation that there is no any traps here :) It looks like the effects I have read about relate not to the DC lines themselves but to the external capacitors installed at the end of a  DC line as a part of some converter schemes.

>  I think what would make sense is to import a "standard linetype" from PyPSA and set the value x and c to zero. But I remember you faced issues with this. Can you describe how we can reproduce the bug? Or post it on PyPSA as an issue?

Agree, it would be the most convenient way. However, if the standard linetype is specified its' parameters would overwrite any manually set values according to the PyPSA [Line documentation ](https://pypsa.readthedocs.io/en/latest/components.html#line) (see description field of the table). I have tried and it worked in full accordance with the documentation :) Actually, it should not be a big problem to add a couple of the DC linetypes. It seems there is no really standard types like `243-AL1/39-ST1A 110.0` or something but there are some common designs and more or less typical `r` and `i_nom` values. I still have collected some data and just needed to be sure that nothing really major was missed.

Regarding the issues, there were some singularity troubles when I have tried to specify the line parameters manually instead of using the linetype specification. It would be very interesting to find the exact reason but first I would prepare a PR for the DC standard linetypes.

**Comment by davide-f** [comment]
In general, we should not forget that the physics behind the modeling are the time equations.
In the case of the inductance: $V = L \frac{dI}{dt}$
All lines are subject to them, however, while in AC we have a sinusoidal v, which induces a sinusoidal I, than that term cannot be neglected.
In DC, obviously variances are common but as we study the so-called steady-state, they are of secondary importance as slowly changing.

So, for DC lines main basically only resistances are of major concerns. That surely implicy the resistance to transmit power, and, eventually, the losses towards the ground as air and isolators do have a high but not infinity resistance. However, this latter term can be neglected, that's why we can focus mainly on the r.

Thank @ekatef for the clarification.
Regarding the manual specification, if you feel to report the issue, we need a reproducible procedure to clarify that, despite following the documentation on the matter:
`Name of line standard type. If this is not an empty string “”, then the line standard type impedance parameters are multiplied with the line length and divided/multiplied by num_parallel to compute x, r, etc. This will override any values set in r, x, and b. If the string is empty, PyPSA will simply read r, x, etc.` (from line documentation)

If you experience the issue again while preparing the PR, you may notify to PyPSA community in github.
